pkg/eks/tasks.go

```diff
@@ -0,0 +1,44 @@
+package eks
+
+import (
+	"github.com/kris-nova/logger"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/cfn/manager"
+)
+
+type clusterConfigTask struct {
+	info string
+	spec *api.ClusterConfig
+	call func(*api.ClusterConfig) error
+}
+
+func (t *clusterConfigTask) Describe() string { return t.info }
+
+func (t *clusterConfigTask) Do(errs chan error) error {
+	err := t.call(t.spec)
+	close(errs)
+	return err
+}
+
+// AppendExtraClusterConfigTasks returns all tasks for updating cluster configuration or nil if there are no tasks
+func (c *ClusterProvider) AppendExtraClusterConfigTasks(cfg *api.ClusterConfig, tasks *manager.TaskTree) {
```

Comment:
Could we leave `Extra` out of the method's name? I think `Append` conveys that information.

```suggestion
func (c *ClusterProvider) AppendClusterConfigTasks(cfg *api.ClusterConfig, tasks *manager.TaskTree) {
```

Reply:
Sort of, but they are all optional, so I'd rather stick to a more explicit name...